### PR TITLE
Use public consumer API for CLI consumer

### LINKED ISSUE #397: CLI: refactor consume to use documented API
Currently, CLI uses undocumented API to perform consumer fetch and display.   We need to convert this to use published API

### PR DESCRIPTION
**Pull request by nicholastmosher**
This closes #397 

- Simplify CLI consumer
- Reduce dependency of CLI consumer on the internal details of API consumer
- Enhance API consumer with any required information

### CODE REVIEW

File: src/client/src/consumer.rs [hunk 1]
```diff
@@ -392,7 +405,13 @@ impl ConsumerConfig {
 
 /// The individual record for a given stream.
 pub struct Record {
```
Comment (sehz) [inline src/client/src/consumer.rs:407]: these are already in the batch header why creating duplicate information?

**Comment by sehz** [comment]
Instead of trying to use a single monolithic api for everything,  create different iterator for different use case.

For example, in the `HashMap`,  there is `values` and `keys`.    Likewise, we can have `batches` which will iterate batches or  `record` which will may have tuple of headers, and records.

File: src/client/src/consumer.rs (hunk 1 above)
Comment (nicholastmosher) [reply src/client/src/consumer.rs:483]: I removed the `partition_index` and `log_start_offset` as they ended up not being needed by the CLI consumer

**Comment by nicholastmosher** [comment]
In the newest update, there is now `stream_with_config` and `stream_batches_with_config` as the basic streaming-consumer functions.

File: src/client/src/consumer.rs [hunk 2]
```diff
@@ -390,9 +437,53 @@ impl ConsumerConfig {
     }
 }
 
+pub struct RecordBatch {
```
Comment (sehz) [inline src/client/src/consumer.rs:440]: Instead of creating another duplicated struct, can you add to Batch in the data plane

**Review by sehz — CHANGES_REQUESTED** [review]

File: src/client/src/consumer.rs (hunk 2 above)
Comment (nicholastmosher) [reply src/client/src/consumer.rs]: Got it. I added new methods `.records()` and `.into_records()` that return iterators over the `DefaultRecord`s inside of the batch.

**Comment by sehz** [comment]
can you rebase on this PR? https://github.com/infinyon/fluvio/pull/769

**Comment by nicholastmosher** [comment]
After #769 is merged I will rebase this one more time so that the diff is clear

**Comment by sehz** [comment]
Please rebase from master since #769 is merged. thanks

**Review by sehz — APPROVED** [review]
LGTM.

**Pull request merged**